### DISCUSSION TITLE
How do I pull (with reset) a PR branch?

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [jesseduffield/lazygit#3617] [Q&A] [answered]
I like to check out a PR branch and review locally. Sometimes when I pull, I get this error, because the author of the PR force-pushed:

```
hint: Diverging branches can't be fast-forwarded, you need to either:
hint: 
hint:   git merge --no-ff
hint: 
hint: or:
hint: 
hint:   git rebase
hint: 
hint: Disable this message with "git config advice.diverging false"
fatal: Not possible to fast-forward, aborting.
```

I can drop to the terminal and run something like `git fetch --all && git reset --hard "origin/$(git rev-parse --abbrev-ref HEAD)"` to reset the branch to the latest commit. But ideally, I'd like to do this from lazygit.

Is there a way for me to somehow reset to the latest commit on the remote PR branch via lazygit?

**Comment by stefanhaller** [comment, marked as answer]
While doing a hard reset onto your upstream works, it's not the recommended way to resolve this. The downside is that if you ever have the situation where the upstream was force-pushed, but you also have local commits that you didn't push yet (not at all an uncommon situation for me), the hard reset will blow away your local commits.

It's much better to rebase onto your upstream instead. You can do this manually by selecting your local branch and hitting `u r`.

However, I recommend to do `git config --global pull.rebase true` once, then every "git pull" will automatically do the rebase for you, and you never have to deal with this problem again.

(Side note: `"origin/$(git rev-parse --abbrev-ref HEAD)"` is a very long-winded way of saying `@{u}`, in case you ever need that for other reasons.)

**Reply by fredrikaverpil** [reply]
Excellent, many thanks!
What a "hidden gem". The `u r` and `u g` was exactly what I was looking for!

**Reply by fredrikaverpil** [reply]
> However, I recommend to do `git config --global pull.rebase true` once, then every "git pull" will automatically do the rebase for you, and you never have to deal with this problem again.

I know about this but I would rather have a failing command here, which would indicate to me that I can't merely just "pull". I can then execute `git pull --rebase` if this is what I still want to do. Or I can choose to take another action. 😄
